tutorials/sphinx_tuto/tensordict_memory.py

```diff
@@ -0,0 +1,145 @@
+#            b: Tensor(shape=torch.Size([10, 10]), device=cuda:0, dtype=torch.float32, is_shared=True)},
+#        batch_size=torch.Size([10]),
+#        device=cuda:0,
+#        is_shared=True)
+#
+# You can check the current device of the :class:`TensorDict` with the ``device``
+# attribute.
+#
+# .. code-block::
+#
+#    >>> print(tensordict.device)
+#    cuda:0
+#
+# The contents of the :class:`TensorDict` can be sent to a device like a PyTorch tensor
+# with :meth:`TensorDict.cuda() <tensordict.TensorDict.cuda>` or
+# :meth:`TensorDict.device(device) <tensordict.TensorDict.device>` with ``device``
+# being the desired device.
+#
+# .. code-block::
+#
+#    >>> tensordict.to(torch.device("cpu"))
+#    >>> print(tensordict)
+#    TensorDict(
+#        fields={
+#            a: Tensor(shape=torch.Size([10]), device=cpu, dtype=torch.float32, is_shared=False),
+#            b: Tensor(shape=torch.Size([10, 10]), device=cpu, dtype=torch.float32, is_shared=False)},
+#        batch_size=torch.Size([10]),
+#        device=cpu,
+#        is_shared=False)
+#    >>> tensordict.cuda()
+#    >>> print(tensordict)
+#    TensorDict(
+#        fields={
+#            a: Tensor(shape=torch.Size([10]), device=cuda:0, dtype=torch.float32, is_shared=True),
+#            b: Tensor(shape=torch.Size([10, 10]), device=cuda:0, dtype=torch.float32, is_shared=True)},
+#        batch_size=torch.Size([10]),
+#        device=cuda:0,
+#        is_shared=True)
+#
+# The :meth:`TensorDict.device <tensordict.TensorDict.device>` method requires a valid
+# device to be passed as the argument. If you want to remove the device from the
+# :class:`TensorDict` to allow values with different devices, you should use the
+# :meth:`TensorDict.clear_device <tensordict.TensorDict.clear_device>` method.
+#
+# .. code-block::
+#
+#    >>> tensordict.clear_device()
+#    >>> print(tensordict)
+#    TensorDict(
+#        fields={
+#            a: Tensor(shape=torch.Size([10]), device=cuda:0, dtype=torch.float32, is_shared=True),
+#            b: Tensor(shape=torch.Size([10, 10]), device=cuda:0, dtype=torch.float32, is_shared=True)},
+#        batch_size=torch.Size([10]),
+#        device=None,
+#        is_shared=False)
+#
+# Memory-mapped Tensors
+# ---------------------
+# ``tensordict`` provides a class :class:`~.MemmapTensor` which allows us to store the
+# contents of a tensor on disk, while still supporting fast indexing and loading of the
+# contents in batches. See the `ImageNet Tutorial <./tensorclass_imagenet.html>`_ for an
+# example of this in action.
+#
+# To convert the :class:`TensorDict` to a collection of memory-mapped tensors, use the
+# :meth:`TensorDict.memmap_ <tensordict.TensorDict.memmap_>`.
+
+# sphinx_gallery_start_ignore
+import warnings
+
+import torch
+from tensordict import TensorDict
+
+warnings.filterwarnings("ignore")
+# sphinx_gallery_end_ignore
+
+tensordict = TensorDict({"a": torch.rand(10), "b": {"c": torch.rand(10)}}, [10])
+tensordict.memmap_()
```

Comment:
A crucial point here is the `memmap_like` method. In general, it should be preferred (as it does not write the anything on the entire array) and allows to build things that are populated a small fraction at a time.
Can we add this?